sdk/nodejs/runtime/mocks.ts

```diff
@@ -0,0 +1,96 @@
+// Copyright 2016-2018, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { deserializeProperties, serializeProperties } from "./rpc";
+import { getProject, getStack, setMockOptions } from "./settings";
+
+const provproto = require("../proto/provider_pb.js");
+const resproto = require("../proto/resource_pb.js");
+const structproto = require("google-protobuf/google/protobuf/struct_pb.js");
+
+export interface Mocks {
+    call(token: string, args: any, provider?: string): Record<string, any>;
+    newResource(type: string, name: string, inputs: any, provider?: string, id?: string): { id: string, state: Record<string, any> };
+}
+
+export class MockMonitor {
+    mocks: Mocks;
+
+    constructor(mocks: Mocks) {
+        this.mocks = mocks;
+    }
+
+    private newUrn(parent: string, type: string, name: string): string {
```

Comment:
Do we not have this function now somewhere else in the public API of the SDK?  I thought we did...